lib/number.js

```diff
@@ -53,12 +58,31 @@ internals.Number = class extends Any {
 
     multiple(base) {
 
-        Hoek.assert(Hoek.isInteger(base), 'multiple must be an integer');
-        Hoek.assert(base > 0, 'multiple must be greater than 0');
+        const isRef = Ref.isRef(base);
+
+        return this._test('multiple', base, (value, state, options) => {
+
+            let divisor;
+            if (isRef) {
+                divisor = base(state.parent, options);
+                if (typeof divisor !== 'number') {
+                    return this.createError('number.ref', { ref: base.key }, state, options);
+                }
+
+                if (divisor === 0) {
+                    return this.createError('number.refZero', { ref: base.key }, state, options);
+                }
+            }
+            else {
+                divisor = base;
+                if ((typeof divisor !== 'number') || (divisor === 0)) {
+                    return this.createError('number.illegalMultiple', { ref: base.key }, state, options);
+                }
+            }
 
-        return this._test('multiple', base, function (value, state, options) {
+            const mult = Math.pow(10, Math.max(internals.precision(value), internals.precision(divisor)));
```

Comment:
Is it a real world problem for existing use cases though ? We don't want accurate modulo, we just want 0 when it's a multiple.